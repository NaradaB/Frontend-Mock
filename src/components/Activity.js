import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  container: {
    fontFamily: "'Roboto', sans-serif",
    fontSize: "1rem",
    color: "#3A3A3A",
    display: "flex",
    flexDirection: "column",
    padding: "3px",
    width: "20rem",
    height: "5rem",
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "white",
    borderStyle: "solid",
    borderWidth: "1px",
    borderRadius: "0.2rem",
    borderColor: "#808080",
    marginLeft: "8px",
  },
  location: { fontSize: "0.8rem" },
});

function Activity(props) {
  const classes = useStyles();
  return (
    <div data-testid="activity" className={classes.container}>
      <div data-testid="activity-name-and-level">
        {props.activity.name} ({props.activity.level})
      </div>
      <div className={classes.location} data-testid="activity-location">
        Location: {props.activity.location}
      </div>
      <div data-testid="activity-start-end">
        {props.activity.start_time} - {props.activity.end_time}
      </div>
    </div>
  );
}

export default Activity;
