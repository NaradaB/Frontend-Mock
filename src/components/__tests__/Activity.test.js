import { render, screen, cleanup } from "@testing-library/react";
import Activity from "../../components/Activity";
import renderer from "react-test-renderer";

let activity = {
  id: 1,
  name: "Diving",
  start_time: "9:00",
  end_time: "10:00",
  level: "Experienced",
  location: "Swimming pool 2",
  facility_id: 84,
  tags: [
    {
      id: 540,
      name: "Swimming",
    },
  ],
};

afterEach(() => {
  cleanup();
});

test("should render Activity component", () => {
  render(<Activity activity={activity}></Activity>);
  const activityElement = screen.getByTestId("activity");
  expect(activityElement).toBeInTheDocument();
});

test("should render correct Activity name", () => {
  render(<Activity activity={activity}></Activity>);
  const activityElement = screen.getByTestId("activity-name-and-level");
  expect(activityElement).toHaveTextContent("Diving (Experienced)");
});

test("should render correct Activity location", () => {
  render(<Activity activity={activity}></Activity>);
  const activityElement = screen.getByTestId("activity-name-and-level");
  expect(activityElement).toHaveTextContent("Diving (Experienced)");
});

test("should render correct Activity start and end times", () => {
  render(<Activity activity={activity}></Activity>);
  const activityElement = screen.getByTestId("activity-start-end");
  expect(activityElement).toHaveTextContent("9:00 - 10:00");
});

test("matches snapshot", () => {
  const tree = renderer
    .create(<Activity activity={activity}></Activity>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
