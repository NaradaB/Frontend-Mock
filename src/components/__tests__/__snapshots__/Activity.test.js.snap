// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container-0-2-1"
  data-testid="activity"
>
  <div
    data-testid="activity-name-and-level"
  >
    Diving
     (
    Experienced
    )
  </div>
  <div
    className="location-0-2-2"
    data-testid="activity-location"
  >
    Location: 
    Swimming pool 2
  </div>
  <div
    data-testid="activity-start-end"
  >
    9:00
     - 
    10:00
  </div>
</div>
`;
